grails-app/controllers/com/unifina/controller/api/StreamApiController.groovy

```diff
@@ -203,4 +204,31 @@ class StreamApiController {
 			date: iso8601.format(status.date),
 		] as JSON)
 	}
+
+	@StreamrApi
+	def deleteDataUpTo(String id) {
+		getAuthorizedStream(id, Operation.WRITE) { Stream stream ->
+			Date date = iso8601cal.parse((String) params.date)
```

Comment:
Handle case where parameter `date` is not provided? E.g. 400 error and typical JSON error payload.